src/vanilla/DanModel/JavaEnum.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace AutoRest.Java.DanModel
+{
+    public class JavaEnum
+    {
+        public const string RelativePackage = "models";
+
+        private readonly IList<JavaEnumValue> values;
+
+        public JavaEnum(string headerCommentText, string package, string enumName)
+        {
+            HeaderCommentText = headerCommentText;
+            Package = package;
+            EnumName = enumName;
+            values = new List<JavaEnumValue>();
+        }
+
+        public JavaEnum AddValue(string valueName, string valueString)
+        {
+            values.Add(new JavaEnumValue(valueName, valueString));
+            return this;
+        }
+
+        public JavaEnum WithMaximumMultipleLineCommentWidth(int maximumMultipleLineCommentWidth)
+        {
+            MaximumMultipleLineCommentWidth = maximumMultipleLineCommentWidth;
+            return this;
+        }
+
+        protected IEnumerable<JavaEnumValue> Values
+        {
+            get { return values; }
+        }
+
+        protected int? MaximumMultipleLineCommentWidth { get; private set; }
+
+        protected string HeaderCommentText { get; }
+
+        protected string Package { get; }
+
+        protected string EnumName { get; }
+
+        protected string GetFilePath()
+        {
+            return Path.Combine(RelativePackage, $"{EnumName}.java");
+        }
+
+        protected Action<JavaMultipleLineComment> HeaderComment()
+        {
+            return (JavaMultipleLineComment comment) =>
+            {
+                comment.Line(HeaderCommentText);
+            };
+        }
+
+        protected Action<JavaMultipleLineComment> EnumTypeComment()
+        {
+            return (JavaMultipleLineComment comment) =>
+            {
+                comment.Line($"Defines values for {EnumName}.");
+            };
+        }
+
+        public virtual JavaFile GenerateJavaFile()
+        {
+            return new JavaFile(GetFilePath())
+                .MaximumMultipleLineCommentWidth(MaximumMultipleLineCommentWidth)
+                .MultipleLineComment(HeaderComment())
+                .Line()
+                .Package(Package)
+                .Line()
+                .Import("com.fasterxml.jackson.annotation.JsonCreator",
+                        "com.fasterxml.jackson.annotation.JsonValue")
+                .MultipleLineComment(EnumTypeComment())
+                .Block($"public enum {EnumName}", (enumBlock) =>
```

Comment:
I think the lambdas are the most compelling part of all this-- part of the difficulty with StringBuilders is that they stopped resembling the shape of the code being generated (all flat).